mobile/lib/common/loading_screen.dart

```diff
@@ -14,58 +17,100 @@ import 'package:go_router/go_router.dart';
 class LoadingScreen extends StatefulWidget {
   static const route = "/loading";
 
-  const LoadingScreen({super.key});
+  final Future<void>? restore;
+
+  const LoadingScreen({super.key, this.restore});
 
   @override
   State<LoadingScreen> createState() => _LoadingScreenState();
 }
 
 class _LoadingScreenState extends State<LoadingScreen> {
+  String message = "Welcome to 10101";
+
   @override
   void initState() {
-    super.initState();
-
-    Future.wait<dynamic>([
+    List<Future<dynamic>> futures = [
       Preferences.instance.hasEmailAddress(),
       Preferences.instance.getOpenPosition(),
       isSeedFilePresent(),
-    ]).then((value) {
+      Preferences.instance.isFullBackupRequired(),
+    ];
+
+    if (widget.restore != null) {
```

Comment:
What's `widget.restore`?

Reply:
That's an optional future passed as argument from the restore wallet screen to the loading screen to wait for the restore to complete before starting the backend.